src/components/Form/Form.js

```diff
@@ -1,28 +1,57 @@
 import React, { Component } from 'react';
 
 class Form extends Component {
+  
+submitForm = event => {
+  event.preventDefault();
 
-  handleSubmitForm = event => {
-    event.preventDefault();
+  const {
+    header,
+    users
+  } = this.props.state;
+  const form = event.target;
+  const formInputName = form.name.value;
+  const formInputSurname = form.surname.value;
+  const formInputLocation = form.location.value;
+  const formInputAge = form.age.value;
+  const randomId = Math.floor(Math.random() * 10000);
+
+  const newState = {
+    header: [...header],
+    users: [
+      ...users,
+      {
+        id: `${randomId}`,
+        name: `${formInputName}`,
+        surname: `${formInputSurname}`,
+        location: `${formInputLocation}`,
+        age: `${formInputAge}`
+      }
+    ]
+  };
+
+
+  this.props.changeState(newState);
+
+  form.reset();
+}
 
-    event.target.reset();  
-  }
 
   render(){
     return (
-      <form autoComplete="off" onSubmit={this.handleSubmitForm}>
+      <form autoComplete="off" onSubmit={this.submitForm}>
         <div className="row">
           <div className="col">
-            <input className="form-control" type="text" />
+            <input className="form-control" minLength="3" name="name" required type="text" />
```

Comment:
dodaj onChange events i hanldery do nich, ustaw wartość tych pól w state tego komponentu, lub wyższego komponentu w zależności gdzie będziesz obsługiwał submit.
Jesli submit pójdzie w tym komponencie, to tutaj trzymaj dane w state 
Jesli submit pojdzie w komponencie wyższym to dane w state rodzica.

Reply:
ok tylko tak naprawdę nie mam potrzeby w sumie trzymać zmieniających się danych, głównie interesuje mnie submit, zatem po co go na onChange?

Reply:
bo taka jest dobra praktyka, trzymać dane w state komponentu a nie w HTMLU 

dlatego na onChange updatujesz state, value inputa wyciągasz ze statu zawsze i na submit w metodzie 
```

onSubmit =() => {
const {inputValue} = this.state;
const {onSubmit} = this.props;

onSubmit(inputValue);
}
```